services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/docker_utils.py

```diff
@@ -53,3 +54,11 @@ async def get_running_containers_count_from_names(
         filters = clean_filters({"name": container_names})
         containers = await docker.containers.list(all=True, filters=filters)
         return len(containers)
+
+
+def get_docker_service_images(compose_spec_yaml: str) -> list[str]:
```

Comment:
Suggestion: maybe it would make more sense to have a set here?

Reply:
good point.